### PR TITLE
Added automation resource

### PR DESCRIPTION
**Pull request by RWejlgaard**
This change corresponds to my pull request on go-zendesk

### CODE REVIEW

**Comment by RWejlgaard** [comment]
Builds are failing since automation support has not yet been merged in go-zendesk

**Comment by nukosuke** [comment]
@RWejlgaard 

Hi, v0.5.0 has been released 🎉 

Now it's available on master branch(#161).
Please rebase or merge master.

File: zendesk/resource_zendesk_automation.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+		}
+		actions = append(actions, m)
+	}
+	fields["action"] = actions
+	return setSchemaFields(d, fields)
+}
+
+// Unmarshal the terraform schema to the Zendesk client object
+func unmarshalAutomation(d identifiableGetterSetter) (client.Automation, error) {
+	trg := client.Automation{}
```
Comment (nukosuke) [inline zendesk/resource_zendesk_automation.go]: `trg` means `trigger`. Please rename variables.

**Comment by tamccall** [comment]
Not sure what is going on, but when i try and run the tests locally i am getting the following errors.

```
± |feature/automation-resource ✓| → go test -v -cover -run TestAcc.* ./...
# github.com/nukosuke/terraform-provider-zendesk/zendesk
zendesk/resource_zendesk_automation.go:17:13: undefined: zendesk.AutomationAPI
zendesk/resource_zendesk_automation.go:21:13: undefined: zendesk.AutomationAPI
zendesk/resource_zendesk_automation.go:25:13: undefined: zendesk.AutomationAPI
zendesk/resource_zendesk_automation.go:29:13: undefined: zendesk.AutomationAPI
zendesk/resource_zendesk_automation.go:79:35: undefined: zendesk.Automation
```

Here is my branch info which looks correct to me.
```
 2019-11-29 15:09:27 ⌚  Andys-MacBook-Pro in ~/workspace/test/terraform-provider-zendesk
± |feature/automation-resource ✓| → gl
* b5686df - (HEAD -> feature/automation-resource, origin/feature/automation-resource) renamed 'trg' to 'automation' (9 hours ago) <Rasmus Wejlgaard>
* 275c433 - removed description field (7 days ago) <Rasmus Wejlgaard>
* 734d380 - implemented automations (7 days ago) <Rasmus Wejlgaard>
* daddb9e - (origin/terraform-v0.12) Fix the key to check if resource e
```

**Comment by RWejlgaard** [comment]
@tamccall Hi, have you pulled the latest go-zendesk? the dependent changes was recently merged

**Comment by RWejlgaard** [comment]
@tamccall I'm unable to reproduce your issue, can you try re running?

**Comment by tamccall** [comment]
@RWejlgaard  sorry for being unresponsive. I have been on holiday. I was able to get the tests to run on this branch by updating the client to the latest version.

Currently seeing this failure when running the acceptance test.

```
=== RUN   TestAccAutomationExample
--- FAIL: TestAccAutomationExample (0.00s)
    testing.go:615: Step 0 error: 3 problems:

        - Argument or block definition required: On /var/folders/3n/2m9krr0j0gbcbw02v755nz240000gn/T/tf-test299730118/main.tf line 12: An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.
        - Argument or block definition required: On /var/folders/3n/2m9krr0j0gbcbw02v755nz240000gn/T/tf-test299730118/main.tf line 13: An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.
        - Argument or block definition required: On /var/folders/3n/2m9krr0j0gbcbw02v755nz240000gn/T/tf-test299730118/main.tf line 14: An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.
```

I am also seeing issues with the trigger automation tests. May need to update this branch with the latest master to make sure those are working correctly. I know it is pretty hard to test these things without an api token.

File: examples/automations.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#   https://developer.zendesk.com/rest_api/docs/support/automations
+
+resource "zendesk_automation" "auto-close-automation" {
+  title  = "Close ticket 4 days after status is set to solved"
+  active = true
+
+  all {
+    field: "status",
+    operator: "is",
+    value: "solved"
```
Comment (tamccall) [inline examples/automations.tf]: i think you can fix the build errors i saw by replacing the `:` with an `=`

File: zendesk/resource_zendesk_automation_test.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/nukosuke/go-zendesk/zendesk"
+	"github.com/nukosuke/go-zendesk/zendesk/mock"
+)
+
+func TestMarshalAutomation(t *testing.T) {
+	expected := zendesk.Automation{
```
Comment (tamccall) [inline zendesk/resource_zendesk_automation_test.go]: maybe add position to this as well?

File: examples/automations.tf (hunk 2 above)
Comment (nukosuke) [reply examples/automations.tf]: and the trailing commas should be removed, i think.

**Comment by tamccall** [comment]
still seeing some errors
```
--- FAIL: TestAccAutomationExample (1.34s)
    testing.go:615: Step 0 error: Check failed: Check 1/4 error: Not found: zendesk_automation.auto-reply-automation in [root]
```

File: zendesk/resource_zendesk_automation_test.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAutomationDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: readExampleConfig(t, "automations.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("zendesk_automation.auto-reply-automation", "title", "Auto Reply Automation"),
```
Comment (tamccall) [inline zendesk/resource_zendesk_automation_test.go]: it seems to be named auto close automation in the example

**Pull request merged**